flink-libraries/flink-table/src/test/scala/org/apache/flink/table/utils/TestFilterableTableSource.scala

```diff
@@ -97,6 +98,15 @@ class TestFilterableTableSource(
 
   override def isFilterPushedDown: Boolean = filterPushedDown
 
+  override def getTableStats: TableStats = {
```

Comment:
If we let the `FilterableTableSource` compute the cardinality, this information will be lost if the source table has valid stats registered in the `TableSourceTable`.